### PR TITLE
Set up lint prettier husky hooks

### PR DESCRIPTION
**Pull request by lutzk**
- configure  base tsConfig and extend them
- configure eslint
- configure prettier
- enable strict type checks
- set up husky pre-commit & pre-push hooks
- fix type errors
- delete jsonfig.json - it was overriding tsconfig in editor

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (cemusta) [inline .husky/pre-commit:1]: this can be done via husky config inside package.json

**Review by cemusta — CHANGES_REQUESTED** [review]
I will push a mini change with my notes

File: src/api/config/sections/AuthSection.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { IsNotEmpty, IsUrl } from 'class-validator';
+import { DotEnv } from '..';
```
[the remark this answers is not in the document]
Comment (lutzk) [reply src/api/config/sections/AuthSection.ts:2]:
```
This feature is something most users may never have to think about; however, if you’ve hit issues under --isolatedModules, TypeScript’s transpileModule API, or Babel, this feature might be relevant.
```

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export

File: .husky/pre-commit (hunk 1 above)
Comment (lutzk) [reply .husky/pre-commit:1]: no not in husky >= 5
its like that now

File: tsconfig.dev.json [hunk 3]
```diff
@@ -1,31 +1,4 @@
...
-      "./node_modules/@types",
-      "./src/types"
-    ]
-  },
-  "include": [
-    "src/**/*.ts",
-    "./*.js"
-  ]
+  "extends": "./tsconfig.base",
+  "exclude": ["src/**/*.spec.ts", "src/**/*.test.ts", "src/test/**/*"]
```
[the remark this answers is not in the document]
Comment (lutzk) [reply tsconfig.dev.json:3]: actually i just did it as i fund it before 😆 
we can configure it however you want / need it :)

[the remark this answers is not in the document]
Comment (lutzk) [reply tsconfig.dev.json]: i did it as u say
moved the exclude to the tsconfig.json

**Review by cemusta — APPROVED** [review]
👍

**Pull request merged**